code/modules/antagonists/heretic/knowledge/ash_lore.dm

```diff
@@ -223,6 +224,9 @@
 		existing_beam_spell.beam_duration *= 0.66 // Faster beams
 		existing_beam_spell.cooldown_time *= 0.66 // Lower cooldown
 
+	var/datum/action/cooldown/spell/aoe/fiery_rebirth = locate() in user.actions
```

Comment:
You aren't locating firey rebirth, you're locating the first AOE spell in their action list
```suggestion
	var/datum/action/cooldown/spell/aoe/fiery_rebirth/fiery_rebirth = locate() in user.actions
```